fs/xfstests.py

```diff
@@ -221,6 +229,90 @@ def setUp(self):
         self.test_mnt = self.params.get('test_mnt', default='/mnt/test')
         self.disk_mnt = self.params.get('disk_mnt', default='/mnt/loop_device')
         self.fs_to_test = self.params.get('fs', default='ext4')
+        self.run_type = self.params.get('run_type', default='distro')
+
+        if self.run_type == 'upstream':
+            prefix = "/usr/local"
+            bin_prefix = "/usr/local/bin"
+            sbin_prefix = "/usr/local/bin"
+
+            if self.detected_distro.name == 'SuSE':
+                # SuSE has /sbin at a higher priority than /usr/local/bin
+                # in $PATH, so install all the binaries in /sbin to make
+                # sure they are picked up correctly by xfstests.
+                #
+                # We still install in /usr/local but binaries are kept in
+                # /sbin
+                bin_prefix = "/sbin"
+                sbin_prefix = "/sbin"
+
+            if self.fs_to_test == "ext4":
+                # Build e2fs progs
+                e2fsprogs_dir = os.path.join(self.teststmpdir, 'e2fsprogs')
+                if not os.path.exists(e2fsprogs_dir):
+                    os.makedirs(e2fsprogs_dir)
+                e2fsprogs_url = self.params.get('e2fsprogs_url')
+                git.get_repo(e2fsprogs_url, destination_dir=e2fsprogs_dir)
+                e2fsprogs_build_dir = os.path.join(e2fsprogs_dir, 'build')
+                if not os.path.exists(e2fsprogs_build_dir):
+                    os.makedirs(e2fsprogs_build_dir)
+                os.chdir(e2fsprogs_build_dir)
+                process.run("../configure --prefix=%s --bindir=%s --sbindir=%s"
+                            % (prefix, bin_prefix, sbin_prefix), verbose=True)
```

Comment:
If both bin_prefix and sbin_prefix are one and the same then a single variable can be used instead of 2

Reply:
Right, I think we can get rid of the duplication. Will do. Thanks for pointing it out